typing/env.ml

```diff
@@ -2198,7 +2209,11 @@ let rec components_of_module_maker
             let addr =
               match decl.val_kind with
               | Val_prim _ -> Lazy_backtrack.create_failed primitive_address_error
-              | _ -> next_address ()
+              | Val_reg _ -> next_address ()
+              | Val_ivar _ | Val_self _ | Val_anc _ ->
+                next_address ()
+              | Val_mut _ ->
+                Misc.fatal_error "Mutable variable found at the structure level"
```

Comment:
I think it would be better to follow the `Val_prim` case and use the `Lazy_backtrack.create_failed` error mechanism here. I'm not completely certain why that mechanism is used, but at a quick glance I think it might give better backtraces by collecting the backtrace eagerly?